src/polycubed/src/cube_tc.cpp

```diff
@@ -324,6 +324,15 @@ int pcn_vlan_push_tag(struct CTXTYPE *ctx, u16 eth_proto, u32 vlan_id) {
   return bpf_skb_vlan_push(ctx, eth_proto, vlan_id);
 }
 
+/* helper to get packet timestamp related to Unix Epoch */
```

Comment:
If I am not wrong, the code in XDP and TC looks the same.
Is this wanted? In this case, shouldn't we write it once, maybe in the base cube?

Reply:
Have you tested the xdp version? There shouldn't be a tstamp field in `struct xdp_md`: https://github.com/torvalds/linux/blob/master/include/uapi/linux/bpf.h#L3754-L3763

Reply:
I decided to comment the `ctx->tstamp` selection since its value changes too many times. To compute the Unix epoch, I need the kernel timer, thus `bpf_ktime_get_ns()`. I cannot rely on `ctx->tstamp`. For those use cases where the timestamp does not have to be Unix epoch, it can still be used thus.